pds_doi_core/util/doi_validator.py

```diff
@@ -0,0 +1,323 @@
+        self._database_obj = DOIDataBase(self.m_default_db_file)
+
+    def get_database_name(self):
+        return(self._database_obj.get_database_name())
+
+    def _check_field_title_duplicate(self, doi: Doi,action, force_flag=False):
+        """ Check if the same title exist already in local database with a DOI.
+            If the same title exist but no DOI minted yet, nothing to be done.
+            If the same title exist and a DOI has been minted, throw an exception if action is not 'release'
+            because we want to allow the 'release' action on the same DOI so the metadata can be updated.
+            If the same title exist and a lidvid exist but action is 'draft' and force_flag is true, it
+            is allowed to provide a way to draft the same label.
+            If no title exist, nothing to be done.
+        """
+        query_criterias = {}
+        query_criterias['title'] = [doi.title]  # The database expect each field to be a list.
+
+        logger.debug(f"query_criterias {query_criterias}")
+        logger.debug(f"doi {doi}")
+        logger.debug(f"doi.related_identifier {doi.related_identifier}")
+
+        # Query database for rows with given title value.
+        columns, rows = self._database_obj.select_latest_rows(query_criterias)
+
+        logger.debug(f"len(rows) {len(rows)}")
+
+        col_index_to_doi     = columns.index('doi')
+        col_index_to_lid     = columns.index('lid')
+        col_index_to_vid     = columns.index('vid')
+        col_index_to_status  = columns.index('status')
+        last_row_index       = len(rows) - 1
+
+        # If there are rows with same title and 'lid' and 'vid' fields exist in database.
+        if len(rows) > 0 and (rows[last_row_index][col_index_to_lid] and rows[last_row_index][col_index_to_vid]):
+            col_index_to_doi     = columns.index('doi')
+            col_index_to_lid     = columns.index('lid')
+            col_index_to_vid     = columns.index('vid')
+            col_index_to_status  = columns.index('status')
+            logger.debug(f"LEN_ROWS: {len(rows)}")
+            for ii in range(len(rows)):
+                logger.debug(f"LID_FIELD: {ii,rows[ii][col_index_to_lid]}")
+                logger.debug(f"VID_FIELD: {ii,rows[ii][col_index_to_vid]}")
+            lidvid = rows[last_row_index][col_index_to_lid] + '::' + rows[last_row_index][col_index_to_vid]
+            logger.debug(f"rows[last_row_index][col_index_to_doi] {rows[last_row_index][col_index_to_doi]}")
+            if doi.related_identifier == lidvid:   # If 'lidvid' field exist already with the same title, throw an exception.
+                col_index_to_title   = columns.index('title')
+
+                logger.debug(f"CHECK_FIELD_TITLE_EXIST_TRUE_LIDVID_EXIST_TRUE: doi: {rows[last_row_index][col_index_to_doi]}, title: '{doi.title}' lidvid: {lidvid} action: {action} force_flag: {force_flag}")
+                if action != 'release':
```

Comment:
Please verify my logic below:

The thought was if the user wishes to release a DOI with the same title, normally the function will reject because a title already exist (say 'draft' the same PDS4 label).  If we see the same title and we look at the action and it is 'release' then it is allowed.